source/symmetric_cipher.c

```diff
@@ -180,27 +180,27 @@ int aws_symmetric_cipher_decrypt(
         return AWS_OP_ERR;
     }
 
-    if (cipher->good) {
+    if (cipher->state == AWS_CIPHER_READY) {
         return cipher->vtable->decrypt(cipher, to_decrypt, out);
     }
 
     return aws_raise_error(AWS_ERROR_INVALID_STATE);
 }
 
 int aws_symmetric_cipher_finalize_encryption(struct aws_symmetric_cipher *cipher, struct aws_byte_buf *out) {
-    if (cipher->good) {
+    if (cipher->state == AWS_CIPHER_READY) {
         int ret_val = cipher->vtable->finalize_encryption(cipher, out);
-        cipher->good = false;
+        cipher->state = AWS_CIPHER_FINALIZED;
```

Comment:
nit: need to handle error vs finalized here. finalize could have failed, and in that case it should remain in error state